### PR TITLE
Test And Test Suite Creation (redesign)

### PR DESCRIPTION
**Pull request by yxl3784**

### CODE REVIEW

File: src/gui/MainWindow.cpp [hunk 1]
```diff
@@ -597,6 +597,41 @@ void MainWindow::prepareUi() {
     });
     verifyValidateMenu->addAction(verifyValidateViewportAct);
 
+    QAction* verificationValidationNewTest = new QAction(tr("Create new test"), this);
+    verificationValidationNewTest->setIcon(QPixmap::fromImage(coloredIcon(":/icons/verifyValidateIcon.png", "$Color-MenuIconVerifyValidate")));
```
Comment (isaacy13) [inline src/gui/MainWindow.cpp]: maybe new icon for each (create new test, remove test, etc..)

File: src/gui/VerificationValidationWidget.cpp [hunk 2]
```diff
@@ -443,6 +488,490 @@ void VerificationValidationWidget::testListSelection(QListWidgetItem* test_click
...
+    v_layout2->addSpacing(10);
+    v_layout2->addWidget(addArgFormBtn);
+    v_layout2->addSpacing(20);
+    v_layout2->addWidget(new QLabel("If for tops write PATH or NAME in arguments"));
+    v_layout2->addSpacing(5);
+    QScrollArea* scroll = new QScrollArea();
+    scroll->setWidgetResizable(true);
+    QWidget* content_widget = new QWidget();
+    argLayout = new QHBoxLayout();
+    for(int i = 0; i < 3; i++){
```
Comment (isaacy13) [inline src/gui/VerificationValidationWidget.cpp]: maybe default should only have one box?

File: src/gui/VerificationValidationWidget.cpp [hunk 3]
```diff
@@ -443,6 +488,490 @@ void VerificationValidationWidget::testListSelection(QListWidgetItem* test_click
...
+    QGroupBox* groupbox1 = new QGroupBox("Main Info");
+    QVBoxLayout* v_layout = new QVBoxLayout();
+    
+    QFormLayout* mainForm = new QFormLayout();
+    testNameInput = new QLineEdit();
+    mainForm->addRow("Test Name: ", testNameInput);
+    testCmdInput = new QLineEdit();
+    mainForm->addRow("Test Command: ", testCmdInput);
+    testCategoryInput = new QLineEdit();
+    mainForm->addRow("Test Category: ", testCategoryInput);
```
Comment (isaacy13) [inline src/gui/VerificationValidationWidget.cpp:821]: test category can be implied from test command

File: src/gui/VerificationValidationWidget.cpp [hunk 4]
```diff
@@ -443,6 +488,490 @@ void VerificationValidationWidget::testListSelection(QListWidgetItem* test_click
...
+    v_layout->addLayout(mainForm);
+
+    v_layout->addSpacing(15);
+    v_layout->addWidget(new QLabel("Select test suite to add test to"));
```
Comment (isaacy13) [inline src/gui/VerificationValidationWidget.cpp:825]: currently able to add test without test suite selected

File: src/gui/VerificationValidationWidget.cpp [hunk 5]
```diff
@@ -443,6 +488,490 @@ void VerificationValidationWidget::testListSelection(QListWidgetItem* test_click
...
+
+void VerificationValidationWidget::addArgForm() {
+    int n = argInputList.size()+1;
+    QString boxTitle = "Argument Input %1";
+    QGroupBox* argField = new QGroupBox(boxTitle.arg(n));
+    QFormLayout* argForm = new QFormLayout();
+    QLineEdit* argInput = new QLineEdit();
+    argForm->addRow("Argument: ", argInput);
+    QCheckBox* isVar = new QCheckBox();
+    argForm->addRow("Has variable: ", isVar);
```
Comment (isaacy13) [inline src/gui/VerificationValidationWidget.cpp:769]: able to infer whether argument has a variable based on if variable textbox is empty or not

File: include/VerificationValidationWidget.h [hunk 6]
```diff
@@ -41,6 +41,10 @@ class VerificationValidationWidget : public QHBoxWidget
     void setStatusBar(QStatusBar* statusBar) { this->statusBar = statusBar; }
     QString getDBConnectionName() const { return dbConnectionName; }
 
+    void showNewTestDialog();
```
Comment (isaacy13) [inline include/VerificationValidationWidget.h:48]: reminder to resolve conflicts with dev

**Review by isaacy13 — CHANGES_REQUESTED** [review]

File: src/gui/MainWindow.cpp (hunk 1 above)
Comment (yxl3784) [reply src/gui/MainWindow.cpp]: No time

File: src/gui/VerificationValidationWidget.cpp (hunk 2 above)
Comment (yxl3784) [reply src/gui/VerificationValidationWidget.cpp]: three was the average number of args

**Review by isaacy13 — APPROVED** [review]

**Pull request merged**